unitary/examples/quantum_rpg/location.py

```diff
@@ -0,0 +1,79 @@
+from typing import Dict, List, Optional, Sequence
+
+import dataclasses
+import unitary.examples.quantum_rpg.encounter as encounter
+import enum
+
+
+class Direction(enum.Enum):
+    NORTH = "north"
+    EAST = "east"
+    SOUTH = "south"
+    WEST = "west"
+    UP = "up"
+    DOWN = "down"
+
+    @classmethod
+    def parse(cls, s: str) -> Optional["Direction"]:
+        """Parses a string as a Direction.
+
+        Allows prefixes, like 'e' to be parsed as EAST.
+        """
+        lower_s = s.lower()
+        for d in Direction:
+            if d.value.startswith(lower_s):
+                return d
+
+
+@dataclasses.dataclass
+class Location:
+    """Dataclass representing a location in the quantum RPG.
+
+    Attributes:
+        label: id of the location so that other rooms can
+            refer to it succintly.
+        title: Title or short description of the room.
+        description: Longer description of the room.
+        exits: Mapping from direction to label of the
+           adjacent rooms.
+        encounter: Sequence of encounters that can be
+           visited here.
+
+    """
+
+    label: str
+    title: str
+    exits: Dict[Direction, str]
```

Comment:
[out of scope of this PR] It might be fun to introduce Quantumness to this dict. Exits can be in superpositions or even entangled, manipulated by the outcomes of encounters in the location. This will allow developers to create trippy and interesting worlds where quantumness is core to the story and in navigating the world itself, rather than limited to interactions in battle/environments.

e.g. kill the evil goo on the lever to flip it and put `Hadamard` on the west exit, allowing you 50% chance of bypassing the maze. Another encounter may allow you to entangle this superposed exit with another superposed `crocodile_pit + safe_house` exit from the maze. Now, if you're unsuccessful in bypassing the maze, the entangled exit collapses to only `safe_house` and you've secured safety for your party either way.

Reply:
Very interesting idea.  I think it's a good idea.  I need to think this through a bit of how to do it.  As you menbtioned, out of scope for this PR (and possibly for the 'MVP' demo).